From bbe4dee58eb22d617a8c63b2e63ee6afebddf521 Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Wed, 5 Dec 2018 05:28:30 +0100
Subject: [PATCH] Merge pull request #311 in SW_UGW/linux from
 bugfix/DRVLIB_SW-1040-dp_bugfix_extvlan to master

* commit '68045d24dfffe50ad44f044bb532722292f6f254':
  Added different Macro to differentiate Ether type 0x8863 and 0x8864
---
 drivers/net/ethernet/lantiq/datapath/gswip31/datapath_tc_asym_vlan.c | 2 ++
 include/net/datapath_api_vlan.h                                      | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_tc_asym_vlan.c b/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_tc_asym_vlan.c
index 74c58e306a59..155947eadaeb 100644
--- a/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_tc_asym_vlan.c
+++ b/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_tc_asym_vlan.c
@@ -645,6 +645,8 @@ static int ext_vlan_cfg(struct core_ops *ops,
 		GSW_EXTENDEDVLAN_FILTER_ETHERTYPE_IPOE,
 		DP_PROTO_PPPOE,
 		GSW_EXTENDEDVLAN_FILTER_ETHERTYPE_PPPOE,
+		DP_PROTO_PPPOE_SES,
+		GSW_EXTENDEDVLAN_FILTER_ETHERTYPE_PPPOE,
 		DP_PROTO_ARP,
 		GSW_EXTENDEDVLAN_FILTER_ETHERTYPE_ARP,
 		DP_PROTO_IP6,
diff --git a/include/net/datapath_api_vlan.h b/include/net/datapath_api_vlan.h
index 64dae7ed0ecb..f88511cc89cd 100644
--- a/include/net/datapath_api_vlan.h
+++ b/include/net/datapath_api_vlan.h
@@ -34,7 +34,8 @@ struct dp_pattern_vlan {
 		  * DP_VLAN_PATTERN_NOT_CARE: don't care
 		  */
 #define DP_PROTO_IP4    ETH_P_IP /*IP packet 0x0800*/
-#define DP_PROTO_PPPOE  ETH_P_PPP_DISC /*PPPoE packet: 0x8863 & 0x8864) */
+#define DP_PROTO_PPPOE  ETH_P_PPP_DISC /*PPPoE packet: 0x8863 */
+#define DP_PROTO_PPPOE_SES ETH_P_PPP_SES /*PPPoE packet: 0x8864 */
 #define DP_PROTO_ARP    ETH_P_ARP /*ARP 0x0806*/
 #define DP_PROTO_IP6    ETH_P_IPV6 /*IPv6 packet 0x86DD*/
 #define DP_PROTO_EAPOL  ETH_P_PAE /*EAPOL packet 0x888E */
